packages/@aws-cdk/aws-apigateway/README.md

```diff
@@ -510,6 +510,66 @@ const api = new apigateway.RestApi(this, 'books', {
   }
 })
 ```
+### Access Logging
+
+This is in addition to the detailed execution logs already provided by Amazon CloudWatch for API requests made to your APIs.
+The access logging feature lets you generate access logs in different formats such as CLF (Common Log Format), JSON, XML, and CSV.
+More info can be found [here](https://docs.aws.amazon.com/apigateway/latest/developerguide/set-up-logging.html).
+
+Access logs can be configured to be sent to a specific CloudWatch log group. This can either be configured either for all stages of the RestApi or to specific stages.
+The following example will configure API Gateway to enable custom access logging.
+
+```ts
+const format = JSON.stringify({
+  requestId: apigateway.AccessLogFormat.contextRequestId(),
+  ip: apigateway.AccessLogFormat.contextIdentitySourceIp(),
+  caller: apigateway.AccessLogFormat.contextIdentityCaller(),
+  user: apigateway.AccessLogFormat.contextIdentityUser(),
+  requestTime: apigateway.AccessLogFormat.contextRequestTime(),
+  httpMethod: apigateway.AccessLogFormat.contextHttpMethod(),
+  resourcePath: apigateway.AccessLogFormat.contextResourcePath(),
+  status: apigateway.AccessLogFormat.contextStatus(),
+  protocol: apigateway.AccessLogFormat.contextProtocol(),
+  responseLength: apigateway.AccessLogFormat.contextResponseLength()
+});
```

Comment:
Thanks for making these changes. I think we need to give this another round of treatment treatment and encapsulate the `JSON.stringify()` here. 

Looking at the examples and the blog posts around this feature, it seems that the most common use case here is to use JSON structure or CLF as the log format. The CDK should make this use case feel more natural. 

For JSON - 

```ts
{
  format: AccessLogFormat.json({
    requestId: AccessLogVariable.contextRequestId(),
    'request_time': AccessLogVariable.contextRequestTime(),
    ...
  });
```

For CLF  -

```ts
{
  format: AccessLogFormat.clf(),
}
```

This, of course, should allow for the possibility that the customer may want a completely different structure to their access log and so should allow - 

```ts
{
  format: AccessLogFormat.custom(`......`);
}
```

Reply:
Once we have factory functions a la `AccessLogFormat.whatever(...)` we can add as many conveniences as we want. For example, I think I'm going to poke my eyes out if I have to write:

```ts
AccessLogFormat.json({
 requestId: apigateway.AccessLogFormat.contextRequestId(),
  ip: apigateway.AccessLogFormat.contextIdentitySourceIp(),
  caller: apigateway.AccessLogFormat.contextIdentityCaller(),
  user: apigateway.AccessLogFormat.contextIdentityUser(),
  requestTime: apigateway.AccessLogFormat.contextRequestTime(),
  httpMethod: apigateway.AccessLogFormat.contextHttpMethod(),
  // ...
});
```

Not to say we shouldn't have the `.json()` method. We *should*, but we *also* should have things like the following:

```ts
AccessLogFormat.jsonWithStandardFields({
  // Long live autocomplete!
  requestId: true,
  ip: true,
  requestTime: false,
});
```

```ts
// We prefix '$context.' to everything to keep it manageable
AccessLogFormat.jsonWithStandardFields('requestId', 'identity.sourceIp', 'requestTime', 'httpMethod');
```

Or whatever we come up with...

Reply:
By the way I think it makes more sense to have the `.json()` factory function and the predefined field names `.contextRequestId()` on different objects.

How about:

```ts
AccessLogFormat.json(...)

AccessLogField.REQUEST_ID
```
?

Reply:
> By the way I think it makes more sense to have the .json() factory function and the predefined field names .contextRequestId() on different objects.

Yip. My suggestion above was `AccessLogFormat` and `AccessLogVariable` but I'm fine with `AccesLogField` as well.